app/lib/slack/event_handler.rb

```diff
@@ -0,0 +1,12 @@
+module Slack
+  class EventHandler
+    def self.call(event_data:)
+      event_class = "Slack::Events::#{event_data[:type]&.classify}".constantize
+
+      event_class.call(event_data)
+    rescue NameError, NotImplementedError
+      Rails.logger.error "Unknown Slack event"
+      false
```

Comment:
I wonder if being more clear about the event types we're handling here would be better? Leaning on `classify` and `constantize` and error handling feels a bit risky to me.

An alternative:

```ruby
case event_data[:type]
when "team_join"
  Slack::Events::TeamJoin.call(event_data)
else
  Rails.logger.debug "Unhandled Slack event: #{event_data[:type]}"
end
```